### ISSUE TITLE
Tests fail with an SSL error on Ubuntu Vivid / Debian Jessie

### ISSUE THREAD

**Issue by cd1** [kimchi-project/kimchi#637]
I'm not able to run the tests on Ubuntu 15.04 Vivid and Debian 8.0 Jessie. Every test fails with an SSL related error.

I'm running Kimchi on Ubuntu 15.04, commit a65075a1a691537a810d1c101b30d501682380f9, but the issue also occurs on Debian 8.0.

## How to reproduce ##

1. Install Kimchi from the Github repository.
2. From the working directory, run `sudo make check`.

## Expected behavior ##

After step (2), the tests should complete successfully.

## Actual behavior ##

All tests fail:

```
FAILED (errors=87, skipped=2)
```

The stack trace of one of the failing functions is:

```
======================================================================
ERROR: test_storagevolume (test_mock_storagevolume.MockStorageVolumeTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_mock_storagevolume.py", line 64, in test_storagevolume
    partitions = json.loads(self.request('/host/partitions').read())
  File "utils.py", line 152, in request
    return _request(conn, path, data, method, headers)
  File "utils.py", line 146, in _request
    conn.request(method, path, data, headers)
  File "/usr/lib/python2.7/httplib.py", line 1048, in request
    self._send_request(method, url, body, headers)
  File "/usr/lib/python2.7/httplib.py", line 1088, in _send_request
    self.endheaders(body)
  File "/usr/lib/python2.7/httplib.py", line 1044, in endheaders
    self._send_output(message_body)
  File "/usr/lib/python2.7/httplib.py", line 888, in _send_output
    self.send(msg)
  File "/usr/lib/python2.7/httplib.py", line 850, in send
    self.connect()
  File "/usr/lib/python2.7/httplib.py", line 1269, in connect
    server_hostname=server_hostname)
  File "/usr/lib/python2.7/ssl.py", line 352, in wrap_socket
    _context=self)
  File "/usr/lib/python2.7/ssl.py", line 579, in __init__
    self.do_handshake()
  File "/usr/lib/python2.7/ssl.py", line 808, in do_handshake
    self._sslobj.do_handshake()
SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)
```

**Comment by ramonmedeiros** [comment]
Hi,

can you please send the python version? Because Python 2.7.9 enabled certificate validation by default for HTTP connections, see PEP476 (https://www.python.org/dev/peps/pep-0476/).

So, if we are running this version, we may ask the user to accept the certificate without verify it or, do not verify certificates by default on Kimchi

**Comment by cd1** [comment]
Thanks for the investigation!

> can you please send the python version?

Both systems are indeed running Python 2.7.9.

> So, if we are running this version, we may ask the user to accept the certificate without verify it or, do not verify certificates by default on Kimchi

There's no actual user in this context. As described above, this issue is related to the automated tests only (i.e. when you run `sudo make check`). Your second suggestion (_do not verify certificates by default on Kimchi_) may be a solution for us, but I guess you meant not to verify the certificates when running the tests - the client side is responsible for verifying certificates, not the server side. In this case, the test infrastructure (i.e. the files inside `tests/`) is acting as the client.

If we could disable the certificate verification while running the tests, I guess it should be enough.

**Comment by ramonmedeiros** [comment]
Thanks Cristian, i will take a look. Please assign to me

**Comment by pvital** [comment]
I guess this problem can be reproducible using the UI also. 
I'm running Kimchi in a Debian-8 server (using phyton 2.7.9) and when I try to connect, using a workstation, to the web console (VNC or Spice) in the browser UI, I'm not able to connect receiving a connection refused error.

Looking the output of Web Inspector when using Spice I got the following error:

```
[Error] WebSocket network error: OSStatus Error -9807: Invalid certificate chain  (spice_auto.html, line 0)
```

**Comment by cd1** [comment]
Thanks for the update! I couldn't find a way to reproduce this issue while running the web UI, I thought it was only triggered during the tests.

**Comment by ramonmedeiros** [comment]
Hi @pvital, 

I'm trying to reproduce your error, but i can't. May i missing some step?

BTW, the patch was sent to fix tests: http://lists.ovirt.org/pipermail/kimchi-devel/2015-May/010524.html

**Comment by pvital** [comment]
@ramonmedeiros don't worry about my problem.
I discovered later that the problem is a VPN restriction that blocks some ports and I could not access the VNC or Spice.

**Comment by cd1** [comment]
Fixed by 205a3f479766b1f83a7c4008ddb14f41074cd79e.
